### DISCUSSION TITLE
Two Important Improvements for TeamTalk Requested

### DISCUSSION THREAD

**Discussion by BrianOlesen02** [BearWare/TeamTalk5#2415] [Q&A]
### Description

Dear TeamTalk Development Team,

I am writing to suggest a couple of improvements that I believe would greatly enhance the user experience for TeamTalk.

Firstly, a modern voice over IP solution should ideally provide stereo sound to all connected clients. Unfortunately, this is not currently the case for TeamTalk. While the Windows (and likely macOS) and Android clients do send and receive stereo sound effectively, the iPhone client does not. I would greatly appreciate it if this issue could be addressed in the near future.

Secondly, I have some concerns regarding the recording options for conferences. Currently, there are two options: recording the entire meeting in one file or saving separate clips for each individual speaker. While the first option works well, the second one is, in my opinion, not very practical. Managing thousands of isolated clips for a single conference is quite cumbersome.

My preferred method for using these recordings is to import them into my DAW (Digital Audio Workstation) and create a separate track for each user. This allows me to mix the entire meeting optimally, removing any poor transmissions and retaining only the clear speech. However, handling such a large number of separate files is challenging, even though they are time-stamped and include the user's name.

A much better solution would be to create a single file for each user that runs as long as the user is logged in. Ideally, this file could include some empty space before the user's first sound to make importing the files easier, although this is not a critical requirement.

Thank you for considering these suggestions. I believe these improvements would significantly enhance the usability of TeamTalk for all users.

Best regards,  
Brian Hovmand Olesen

### Application

- [ ] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkServer

**Comment by peter1384** [comment]
hello
Your first suggestion is already being considered, And it will be implemented at some point
about your second suggestion, you can already cut the empty spaces of a recoreded file by reaper or some other audio editing app, and do whatever you want with it, and also, you can record teamtalk by another recorder app, just enough if you set the teamtalk's output device to the same as your recorder's input device

**Comment by BrianOlesen02** [comment]
Hi  Peter,

Sounds really great. It has been possible since iOS 13 I think, and I’ve pointed this out to Bjørn for years. 😊 If you use TeamTalk for music or game transmition stereo is quite nice to have, and it’s sad the many blind iphone users out there are not able to do that.

 

 

Ah with many participants thousands of files are not apropriate to work with in Reaper.

Another recording program can not split the users so it’s not an option.

 

Brian

 

Fra: peter1384 ***@***.***> 
Sendt: 13. juni 2024 19:11
Til: BearWare/TeamTalk5 ***@***.***>
Cc: BrianOlesen02 ***@***.***>; Author ***@***.***>
Emne: Re: [BearWare/TeamTalk5] Two Important Improvements for TeamTalk Requested (Issue #2374)

 

hello
Your first suggestion is already being considered, And it will be implemented at some point
about your second suggestion, you can already cut the empty spaces of a recoreded file by reaper or some other audio editing app, and do whatever you want with it, and also, you can record teamtalk by another recorder app, just enough if you set the teamtalk's output device to the same as your recorder's input device

**Comment by peter1384** [comment]
you can select the "Separate audio file for each user" option in "Record Conversations to Disk" dialog in teamtalk, and you can manage the recorded files by another app as how as you want
Be pationt, your first suggestion will be implemented when there is enough time, or when someone who knows how to do it makes a contribution.

**Comment by BrianOlesen02** [comment]
Hi again, Peter,

I hope you're doing well.

I wanted to reach out regarding the current situation with our conferences. We have several conferences with many participants, resulting in thousands of files, each containing a small voice clip. As it stands, a new file is created every time a user speaks.

The feature “Separate audio file for each user” suggests that there should be one file per user, not multiple files. This discrepancy is quite confusing, as one would expect a single file per user. I recall that it worked differently in previous versions.

This issue has been a bit disappointing and has forced us to switch to a Zoom pro account to find a workable solution. 

Despite this, I want to acknowledge the great features you’ve introduced over the years, like the Danish interface and the fine opus encoder. These improvements are much appreciated.

We've been using TeamTalk for about 20 years and would love to continue, but it's crucial that user feedback is taken seriously. We hope TeamTalk can be brought up to standard to compete with other VoIP solutions out there.

Best regards,

Brian

 

Fra: peter1384 ***@***.***> 
Sendt: 13. juni 2024 19:51
Til: BearWare/TeamTalk5 ***@***.***>
Cc: BrianOlesen02 ***@***.***>; Author ***@***.***>
Emne: Re: [BearWare/TeamTalk5] Two Important Improvements for TeamTalk Requested (Issue #2374)

 

you can select the "Separate audio file for each user" option in "Record Conversations to Disk" dialog in teamtalk, and you can manage the recorded files by another app as how as you want
Be pationt, your first suggestion will be implemented when there is enough time, or when someone who knows how to do it makes a contribution.

**Comment by peter1384** [comment]
oh really? if a new file is being created everytime the user start to talk then it is really bad and it should be fixed, a new file should only be created when the user logs out and back in again I guess.
But ok, your issue will be kept track of, and it will be resolved as soon as possible. Thanks for helping teamtalk project by your suggestions, reports, and opinions.

**Comment by BrianOlesen02** [comment]
HI Peter,

Yes that is the case. Good I explained it better this time.

You should try it out yourself.

I got quite a shock the first time I did this and saw all these files. 😊

 

Brian

 

 

Fra: peter1384 ***@***.***> 
Sendt: 14. juni 2024 21:41
Til: BearWare/TeamTalk5 ***@***.***>
Cc: BrianOlesen02 ***@***.***>; Author ***@***.***>
Emne: Re: [BearWare/TeamTalk5] Two Important Improvements for TeamTalk Requested (Issue #2374)

 

oh really? if a new file is being created everytime the user start to talk then it is really bad and it should be fixed, a new file should only be created when the user logs out and back in again I guess.
But ok, your issue will be kept track of, and it will be resolved as soon as possible. Thanks for helping teamtalk project by your suggestions, reports, and opinions.

**Comment by Simon818** [comment]
iOS is stereo both ways, you just need to turn off voice preprocessing. Am I misunderstanding?

**Comment by BrianOlesen02** [comment]
Hi,

Are you 100 % totally sure of that?

Please try it yourself, cause I can’t get it to work

 

Fra: Simon Jaeger ***@***.***> 
Sendt: 23. juli 2024 16:11
Til: BearWare/TeamTalk5 ***@***.***>
Cc: BrianOlesen02 ***@***.***>; Author ***@***.***>
Emne: Re: [BearWare/TeamTalk5] Two Important Improvements for TeamTalk Requested (Issue #2374)

 

iOS is stereo both ways, you just need to turn off voice preprocessing. Am I misunderstanding?

**Comment by Simon818** [comment]
@BrianOlesen02 I stream in stereo multiple times a week. It definitely works. I've had occasional problems with the stereo microphones on the iPhone coming through as mono, but any external audio devices always work fine. So yes, I'm sure stereo works both ways, and pretty much always has. Is it possible you have voice preprocessing turned on?

**Comment by BrianOlesen02** [comment]
Hi,

Ok. Using an iPhone?

Cause it works great from android and a pc. I’ll try later today to see if voice preprocessing is turned on, but the iphone doesn’t receive in stereo either, and that shouldn’t have anything to do with voice preprocessing.

 

 

Fra: Simon Jaeger ***@***.***> 
Sendt: 23. juli 2024 23:55
Til: BearWare/TeamTalk5 ***@***.***>
Cc: BrianOlesen02 ***@***.***>; Mention ***@***.***>
Emne: Re: [BearWare/TeamTalk5] Two Important Improvements for TeamTalk Requested (Issue #2374)

 

@BrianOlesen02 <https://github.com/BrianOlesen02>  I stream in stereo multiple times a week. It definitely works. Is it possible you have voice preprocessing turned on?

**Reply by Simon818** [reply]
Voice preprocessing certainly does affect whether you receive in stereo, as turning it on will cause your phone to use a completely different audio system commonly used on phone calls. Everything about the audio in both directions is different. Please do let us know if you find that stereo is not working with voice preprocessing turned off though.
